src/dbnode/storage/index/block.go

```diff
@@ -765,10 +769,39 @@ func (b *block) segmentsWithRLock() []segment.Segment {
 // results datastructure is used to copy it every time documents are added
 // to the results datastructure).
 func (b *block) Query(
+	ctx context.Context,
 	cancellable *resource.CancellableLifetime,
 	query Query,
 	opts QueryOptions,
 	results BaseResults,
+) (bool, error) {
+	ctx, sp := ctx.StartTraceSpan(tracepoint.BlockQuery)
+	sp.LogFields(
```

Comment:
Here too, don’t think this has any more than the parent

Reply:
How about we just remove the spans in `execBlockAggregateQueryFn` and `execBlockQueryFn` and keep these ones inside the block themselves?

It seems like it doesn't add much to have those ones as they are just wrappers to call into these.